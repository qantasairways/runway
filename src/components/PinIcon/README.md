```js
<PinIcon color="#656565" />
```
