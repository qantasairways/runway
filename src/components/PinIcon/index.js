import React from 'react';
import PropTypes from 'prop-types';
import { layout } from '../../theme/airways';

function PinIcon({ color }) {
  return (
    <svg
      focusable="false"
      width={layout.iconSize}
      height={layout.iconSize}
      viewBox="0 0 24 24"
    >
      <defs>
        <path
          id="a"
          d="M7 0C3.13 0 0 3.13 0 7c0 5.25 7 13 7 13s7-7.75 7-13c0-3.87-3.13-7-7-7zm0 9.5a2.5 2.5 0 0 1 0-5 2.5 2.5 0 0 1 0 5z"
        />
      </defs>
      <g fill="none" fillRule="evenodd" transform="translate(5 2)">
        <mask id="b" fill="#fff">
          <use xlinkHref="#a" />
        </mask>
        <g fill={color} mask="url(#b)">
          <path d="M-5-2h24v24H-5z" />
        </g>
      </g>
    </svg>
  );
}

PinIcon.propTypes = {
  color: PropTypes.string.isRequired
};

export default PinIcon;
