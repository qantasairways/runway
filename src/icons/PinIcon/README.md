```js
<PinIcon height="50" width="50" color="#000" />
```
