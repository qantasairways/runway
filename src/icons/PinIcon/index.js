import React from 'react';
import PropTypes from 'prop-types';
import { generate } from 'shortid';

function PinIcon({ height, width, className }) {
  const pathId = generate();
  const maskId = generate();

  return (
    <svg
      width={width}
      height={height}
      className={className}
      focusable="false"
      viewBox="0 0 24 24"
    >
      <defs>
        <path
          id={pathId}
          d="M7 0C3.13 0 0 3.13 0 7c0 5.25 7 13 7 13s7-7.75 7-13c0-3.87-3.13-7-7-7zm0 9.5a2.5 2.5 0 0 1 0-5 2.5 2.5 0 0 1 0 5z"
        />
      </defs>
      <g fillRule="evenodd" transform="translate(5 2)">
        <mask id={maskId} fill="#fff">
          <use xlinkHref={`#${pathId}`} />
        </mask>
        <g mask={`url(#${maskId})`}>
          <path d="M-5-2h24v24H-5z" />
        </g>
      </g>
    </svg>
  );
}

PinIcon.propTypes = {
  height: PropTypes.string,
  width: PropTypes.string,
  className: PropTypes.string
};

PinIcon.defaultProps = {
  height: '24',
  width: '24',
  className: ''
};

export default PinIcon;
