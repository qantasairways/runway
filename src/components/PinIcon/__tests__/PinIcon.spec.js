import React from 'react';
import { mount } from 'enzyme';
import PinIcon from '..';

describe('PinIcon', () => {
  let component;

  it('renders correctly with defaults', () => {
    component = mount(<PinIcon color="hotpink" />);

    expect(component).toMatchSnapshot();
  });
});
