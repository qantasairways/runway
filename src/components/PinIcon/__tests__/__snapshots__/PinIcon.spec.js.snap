// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PinIcon renders correctly with defaults 1`] = `
<PinIcon
  color="hotpink"
>
  <svg
    focusable="false"
    height="32px"
    viewBox="0 0 24 24"
    width="32px"
  >
    <defs>
      <path
        d="M7 0C3.13 0 0 3.13 0 7c0 5.25 7 13 7 13s7-7.75 7-13c0-3.87-3.13-7-7-7zm0 9.5a2.5 2.5 0 0 1 0-5 2.5 2.5 0 0 1 0 5z"
        id="a"
      />
    </defs>
    <g
      fill="none"
      fillRule="evenodd"
      transform="translate(5 2)"
    >
      <mask
        fill="#fff"
        id="b"
      >
        <use
          xlinkHref="#a"
        />
      </mask>
      <g
        fill="hotpink"
        mask="url(#b)"
      >
        <path
          d="M-5-2h24v24H-5z"
        />
      </g>
    </g>
  </svg>
</PinIcon>
`;
